### PR TITLE
Enhancements and fixes related to logging, metrics, and testing

### LINKED ISSUE #5751: Missing metric for current forging mode on a blockproducer node
With the push to P2P also on the BlockProducer node, SPOs need to change there backup/failover infrastructure.

A test with node 8.9.1 showed, turning the blockproduction off on a node via a SIGHUP signal (and removed credential files), all last metrics of the blockproducer stays. So there is currently no way to detect the forging mode of a running node that way.

We would need an additional metric (accessable via prometheus interface or a cli query) with value 1 = forging and 0 = not forging, to be sure we know what the node is currently doing. And the backup/failover infra can handle the workmode accordingly.

Trying to get this data out of the logfiles is not a nice way, and i am sure we can add this metric to all the other available values.

### PR DESCRIPTION
**Pull request by jutaro**
#### **Summary**
This pull request addresses multiple enhancements and fixes related to logging, metrics, and testing in the project.

---

#### **Changes**
1. **Let the Forging Mode metrics respect the command line config**  
   - Resolves [#5751](https://github.com/project/repo/issues/5751).  
   - Fixes a metric to track the **current forging mode** on a block producer node.  
   - Ensures visibility into the forging mode to improve monitoring and debugging.

2. **Enhanced Logging for Chain Events**  
   - Added the following fields to log messages for better traceability:  
     - **`tipBlockHash`**: Hash of the current tip block.  
     - **`tipBlockParentHash`**: Hash of the parent block of the tip.  
     - **`tipBlockIssuerVerificationKeyHash`**: Hash of the verification key of the tip block issuer.  
   - Affected log messages:  
     - `SwitchedToAFork`  
     - `AddedToCurrentChain`

3. **Consistency Testing for Mainnet Configuration**  
   - Added a **test case** to ensure **mainnet configuration consistency**.  
   - Validates the correctness of the mainnet configuration to prevent misconfigurations.

#### **Testing**
- Verified the new metric appears correctly in monitoring systems during runtime.  
- Confirmed the additional fields are included in the logs for the relevant events (`SwitchedToAFork`, `AddedToCurrentChain`).  
- Successfully ran the new mainnet configuration test case.

---
#### **Checklist**
- [x] Corrected metric for current forging mode.  
- [x] Updated log messages with additional fields.  
- [x] Implemented and verified the mainnet configuration consistency test.  
- [x] Updated relevant documentation if necessary.  

---

#### **Linked Issue**
Closes #5751

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 1]
```diff
@@ -493,8 +504,18 @@ instance ( LogFormatting (Header blk)
                | dtal == DDetailed ]
             ++ [ "events" .= toJSON (map (forMachine dtal) events)
                | not (null events) ]
+            ++ [ "tipBlockHash" .= tipBlockHash]
+            ++ [ "tipBlockParentHash" .= tipBlockParentHash]
+            ++ [ "tipBlockIssuerVerificationKeyHash" .= tipBlockIssuerVkHashText]
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:509]: Can you use a 3-element list here (or add those fields to the non-conditional list above), avoiding spurious uses of `(++)`?

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 2]
```diff
@@ -506,6 +527,10 @@ instance ( LogFormatting (Header blk)
...
+
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:533]: same as above

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 3]
```diff
@@ -2097,22 +2121,44 @@ data ChainInformation = ChainInformation
...
     }
+  where
+    tipIssuerVkHash :: BlockIssuerVerificationKeyHash
+    tipIssuerVkHash =
+      case AF.head frag of
+        Left AF.AnchorGenesis ->
+          NoBlockIssuer
+        Left (AF.Anchor _s _h _b) ->
+          NoBlockIssuer
+        Right blk -> getIssuerVerificationKeyHash blk
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:2161]: Can the `Left` cases be combined into one? (I'm aware the code is copied from legacy tracing, but it seems redundant).

File: cardano-node/src/Cardano/Node/Run.hs [hunk 4]
```diff
@@ -724,12 +724,13 @@ updateBlockForging startupTracer blockType nodeKernel nc = do
...
           traceWith startupTracer
-                    (BlockForgingUpdate (bool EnabledBlockForging
-                                              DisabledBlockForging
-                                              (null blockForging)))
-
-          setBlockForging nodeKernel blockForging
+                    (BlockForgingUpdate (if isNonProducing || null blockForging
+                                          then DisabledBlockForging
+                                          else EnabledBlockForging))
+          unless isNonProducing $
```
[the remark this answers is not in the document]
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Run.hs:732]: Without unless

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs (hunk 1 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:509]: Ok

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs (hunk 2 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:533]: OK

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs (hunk 3 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:2161]: Done

**Comment by TerminadaDrep** [comment]
Would it be difficult to also enable logging of the reason for "SwitchedToAFork"?  IE: Longer chain vs Equal length but terminal block had lower VRF?
That would close [feature request 6014](<https://github.com/IntersectMBO/cardano-node/issues/6014>)

**Comment by mgmeier** [comment]
@TerminadaDrep I've looked at the issue, and the data traced by `SwitchedToAFork` would be sufficient to give an underlying cause as to why a certain chain candidate won the Praos ordering over the previous one.
However, determining that cause given the old and new chain selection data needs to be implemented, and is outside the scope of this PR. We'll address it in a future one.

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 5]
```diff
@@ -680,7 +745,14 @@ instance MetaTrace  (ChainDB.TraceAddBlockEvent blk) where
         , ( "epoch"
           , "In which epoch is the tip of the current chain."
           )
+        , ( "forks"
+          , "counter for forks"
+          )
+        , ( "tipBlock"
+          , "Hash vallues for tipBlockHash, tipBlockParentHash and tipBlockIssuerVkHashText"
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:752]: * typo `values`
* `tipBlockIssuerVkHashText` is not the field actual field name, it's the name you chose for the let binding. Should be `tipBlockIssuerVKeyHash` to match the trace msg.

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs [hunk 6]
```diff
@@ -703,6 +775,9 @@ instance MetaTrace  (ChainDB.TraceAddBlockEvent blk) where
         , ( "epoch"
           , "In which epoch is the tip of the current chain."
           )
+        , ( "tipBlock"
+          , "Hash vallues for tipBlockHash, tipBlockParentHash and tipBlockIssuerVkHashText"
+          )
```
Comment (mgmeier) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:780]: ^^ same as above

**Review by mgmeier — CHANGES_REQUESTED** [review]

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs (hunk 5 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs]: Fixed

File: cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs (hunk 6 above)
Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/ChainDB.hs:780]: Fixed

**Review by mgmeier — APPROVED** [review]
LGTM, thank you @jutaro

**Pull request merged**